darwin/torch/dataset.py

```diff
@@ -1,66 +1,68 @@
 import json
+import multiprocessing as mp
 from pathlib import Path
 from typing import List, Optional
 
 import numpy as np
+import torch.utils.data as data
 
 import darwin.torch.transforms as T
-from darwin.torch.utils import (
-    convert_polygon_to_sequence,
-    load_pil_image,
-    polygon_area,
-)
-
-class Dataset(object):
-    def __init__(self, dataset, split: Path, transforms: Optional[List] = None):
+from darwin.torch.utils import convert_polygon_to_sequence, load_pil_image, polygon_area
+
+
+class Dataset(data.Dataset):
+    def __init__(self, root: Path, split: Path, transform: Optional[List] = None):
         """ Creates a dataset
 
         Parameters
         ----------
-        dataset : TODO
-            Dataset to load
+        root : Path
+            Path to the location of the dataset on the file system
         split : Path
             Path to the *.txt file containing the list of files for this split.
-        transforms : list[torchvision.transforms]
+        transform : list[torchvision.transforms]
             List of PyTorch transforms
         """
-        self.dataset = dataset
+        self.root = root
         self.split = split
-        self.transforms = transforms
+        self.transform = transform
         self.images_path = []
         self.annotations_path = []
         self.classes = None
         self.original_classes = None
         self.original_images_path = None
         self.original_annotations_path = None
+        self.convert_polygons = None
 
-        #Compose the transform if necessary
-        if self.transforms is not None:
-            self.transforms = T.Compose(transforms)
+        # Compose the transform if necessary
+        if self.transform is not None and isinstance(self.transform, list):
+            self.transform = T.Compose(transform)
 
         # Populate internal lists of annotations and images paths
         if not self.split.exists():
-            raise FileNotFoundError(f"Could not find partition {self.split}"
-                                    f" in {self.dataset.local_path}.")
-        extensions = ["jpg", "jpeg", "png"]
+            raise FileNotFoundError(f"Could not find partition {self.split}" f" in {self.root}.")
```

Comment:
Technically self.split doesn't need to exist in root, so the error message is confusing. 
I would add a line somewhere on how to generate the split (in the docstring)

Reply:
fixed. The root part was actually wrong there.